### PR TITLE
#395 Update the quantile calculator to smooth the fee curve

### PR DESCRIPTION
**Pull request by mudiali**
Added a new factor parameter to make sure that the current quantile does not deviate too much from the previous quantile value.

### CODE REVIEW

File: lib/bitcoin/BitcoinProcessor.ts [hunk 1]
```diff
@@ -107,6 +107,7 @@ export default class BitcoinProcessor {
     this.quantileCalculator = new SlidingWindowQuantileCalculator(BitcoinClient.convertBtcToSatoshis(1),
       ProtocolParameters.windowSizeInGroups,
       ProtocolParameters.quantileMeasure,
+      ProtocolParameters.quantileDeviationWindowFactor,
```
Comment (thehenrytsai) [inline lib/bitcoin/BitcoinProcessor.ts]: Would change factor to percentage be easier to comprehend? e.g.
`maxQuantileDeviationPercentage`, or `maxQuantileChangePercentage`, or `maxSubsequentQuantileChangePercentage`?

Comment (mudiali) [reply lib/bitcoin/BitcoinProcessor.ts]: I was thinking about that too but wanted to stick with Arvind's original suggestions (and also not do the math :) ...) but I agree and will update.

File: lib/bitcoin/fee/SlidingWindowQuantileCalculator.ts [hunk 2]
```diff
@@ -30,12 +30,14 @@ export default class SlidingWindowQuantileCalculator {
...
    * The value to use as an input to the ValueApproximator.
    */
-  private readonly feeApproximate: number = 1.1;
+  private readonly feeApproximate: number = 1.414;
   /**
    * Size of a frequency vector; 1 + max normalized value
    */
   private frequencyVectorSize: number;
 
+  private maxQuantileDeviationPercentageValue: number;
```
Comment (thehenrytsai) [inline lib/bitcoin/fee/SlidingWindowQuantileCalculator.ts]: I guess you took it out as a member variable for testing purposes.

File: lib/bitcoin/models/ProtocolParameters.ts [hunk 3]
```diff
@@ -15,8 +15,8 @@ export default interface ProtocolParameters {
   /** Quantile measure we use for proof of fee; e.g., 0.5 would be the median */
   quantileMeasure: number;
 
-  /** Factor used to calculate how much the quantile values allowed to deviate from the previous value */
-  quantileDeviationWindowFactor: number;
+  /** The max amount that a quantile value are allowed to deviate from the previous value */
```
Comment (thehenrytsai) [inline lib/bitcoin/models/ProtocolParameters.ts]:
```suggestion
  /** The max amount that a quantile value is allowed to deviate from the previous value */
```

File: lib/bitcoin/fee/SlidingWindowQuantileCalculator.ts (hunk 2 above)
Comment (mudiali) [reply lib/bitcoin/fee/SlidingWindowQuantileCalculator.ts]: That and I wanted to divide by 100 and save the value as member.

**Pull request merged**